ai/AI_working_with_ten.md

```diff
@@ -0,0 +1,898 @@
+# TEN Framework Development Guide
+
+**Purpose**: Complete onboarding guide for developing with TEN Framework v0.11+
+**Last Updated**: 2025-10-28
+
+---
+
+## Table of Contents
+
+1. [Framework Overview](#framework-overview)
+2. [Environment Setup](#environment-setup)
+3. [Building and Running](#building-and-running)
+4. [Creating Extensions](#creating-extensions)
+5. [Graph Configuration](#graph-configuration)
+6. [Debugging](#debugging)
+7. [Remote Access](#remote-access)
+8. [Common Issues](#common-issues)
+
+---
+
+## Framework Overview
+
+### What is TEN Framework?
+
+TEN (Temporal Event Network) is a graph-based AI agent framework that connects modular extensions (nodes) through defined data flows (connections). Version 0.11+ uses the `ten_runtime` Python API.
```

Comment:
TEN stands for Transformative Extensions Network, is a voice agent framework to create conversational AI.
(See https://theten.ai/docs)